### PR TITLE
Adds Cypress E2E tests and GitHub Action Cypress workflow

### PR DESCRIPTION
**Pull request by ftianli-amzn**
*Issue #, if available:*

*Description of changes:*
 1. Adds Cypress E2E tests, which contains the following test cases:
 - Monitor:
   - Create a monitor
   - Update a monitor by changing the name
   - Delete a monitor
   - Search a monitor
   - Add a trigger to the monitor
   - Add an action to the trigger
 - Destination:
   - Create a destination
   - Update a destination by changing the name
   - Delete a destination
   - Search a destination
 - Alert:
   - Alert can be in `Active` state
   - Alert can be in `Acknowledged` state
   - Alert can be in `Completed` state
   - Alert can be in `Error` state
   - Alert can be in `Deleted` state
 2. Adds GitHub Action workflow to run E2E tests on every push into `master` branch

References of the code changes: 
1. https://github.com/opendistro-for-elasticsearch/index-management-kibana-plugin/pull/80
1. https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/pull/172
1. https://github.com/opendistro-for-elasticsearch/index-management-kibana-plugin/pull/139

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ftianli-amzn** [comment]
result for `yarn test:jest`:
```
Test Suites: 6 skipped, 83 passed, 83 of 89 total
Tests:       20 skipped, 385 passed, 405 total
Snapshots:   135 passed, 135 total
Time:        54.053 s
Ran all test suites.
✨  Done in 55.34s.
```

result for `yarn run cypress run`:
```
       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  alert_spec.js                            02:42        5        5        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  destination_spec.js                      00:38        4        4        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  monitor_spec.js                          00:47        6        6        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        04:08       15       15        -        -        -  

✨  Done in 287.06s.
```

**Comment by ftianli-amzn** [comment]
Last run before changing the trigger of the workflow from `pull request` event to `push`:
https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/runs/1665755508

**Review by skkosuri-amzn — APPROVED** [review]

File: .github/workflows/cypress-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        run: |
+          cd kibana
+          yarn start --no-base-path --no-watch &
+          sleep 300
+        # timeout 300 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:5601/api/status)" != "200" ]]; do sleep 5; done'
+      - name: Run Cypress tests
+        uses: cypress-io/github-action@v2
+        with:
+          working-directory: kibana/plugins/alerting-kibana-plugin
+          command: yarn run cypress run
```
Comment (qreshi) [inline .github/workflows/cypress-workflow.yml:79]: Let's add `wait-on: 'http://localhost:5601'` after the command entry to ensure the endpoint is available before the tests.

Also, if `wait-on` is added does the `sleep 300` need to be included after `yarn start` above or was that added for another reason?

Comment (ftianli-amzn) [reply .github/workflows/cypress-workflow.yml:79]: There is a doubt if `wait-on` works in this situation, according to some past workflow runs.

Workflow runs with the first test case passed:
https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/467761518/workflow
`sleep 300` and `wait-on` both added (the `wait-on` timeout is 60 seconds by default).

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/470299602/workflow
`Sleep 300` and without `wait-on`


Workflow runs with the first test case failed:
https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/469631264/workflow
`Sleep 240` and `wait-on` both added.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/469535842/workflow
`Sleep 180` and `wait-on` both added.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/467803060/workflow
`Sleep 120` and `wait-on` both added.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/467679972/workflow
`Wait-on` with a time-out for 360 seconds.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/467715838/workflow
`Wait-on` with a time-out for 480 seconds.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/470232057/workflow
Waiting using command "npx wait-on --timeout 30000 http-get://localhost:5601/app/opendistro-alerting"

Comment (ftianli-amzn) [reply .github/workflows/cypress-workflow.yml:79]: As a reference,
here is the "yarn command" for AD Kibana plugin to run the cypress test:
```
"test:e2e": "CHOKIDAR_USEPOLLING=1 CYPRESS_responseTimeout=180000 WAIT_ON_TIMEOUT=900000 start-server-and-test 'cd ../../ && yarn start --no-base-path --oss && cd plugins/anomaly-detection-kibana-plugin' http-get://localhost:5601/app/opendistro-anomaly-detection-kibana 'echo sleeping to wait for server to get ready && sleep 180 && yarn cy:run'"
```
(https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/package.json)

Comment (ftianli-amzn) [reply .github/workflows/cypress-workflow.yml:79]: The `sleep 300` was added to ISM Kibana plugin when upgrading to ES 7.9.1
https://github.com/opendistro-for-elasticsearch/index-management-kibana-plugin/commit/c4357005ebcb79b66bcde4540f725d2276104505

Comment (qreshi) [reply .github/workflows/cypress-workflow.yml:79]: I see, we can go ahead with these changes then, thanks for the links.

**Review by qreshi — APPROVED** [review]

File: .github/workflows/cypress-workflow.yml (hunk 1 above)
Comment (ftianli-amzn) [reply .github/workflows/cypress-workflow.yml:79]: No problem. I will update if I find a better way after experiment in the future. 😁

**Review by lezzago — APPROVED** [review]

**Pull request merged**